### PR TITLE
Cairo support

### PR DESCRIPTION
**Pull request by ringlej**
This implements all of the script_ops in the scenic -> driver interface.

### CODE REVIEW

**Comment by ringlej** [comment]
I added a few commits to fix issues in cairo I encountered while running cairo on our device with a modified version of @axelson hello_scenic_full
image
image

**Comment by ringlej** [comment]
There are now only 6 unimplemented functions in cairo_sciprt_ops.c (PR comment updated)

**Comment by ringlej** [comment]
I've now added some commits that allow building cairo to run on a rpi (tested on rpi3b)

**Comment by ringlej** [comment]
https://github.com/ScenicFramework/scenic_driver_local/assets/100756/6a85af34-cb98-4585-b4f1-d1f0283de84a

https://github.com/ScenicFramework/scenic_driver_local/assets/100756/184abe9b-ec37-4aaa-b1d8-5c404871f718

https://github.com/ScenicFramework/scenic_driver_local/assets/100756/0d4dfffe-a17b-4880-8e72-538a0a8fb907

https://github.com/ScenicFramework/scenic_driver_local/assets/100756/5a9e6e14-5622-443f-bb18-8a422536e5cc

https://github.com/ScenicFramework/scenic_driver_local/assets/100756/103a997a-5e7b-4ff9-bef3-7b4cd10f1417

File: c_src/device/cairo.c [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+  p_ctx->ratio = 1.0f;
+  p_ctx->dist_tolerance = 0.1f * p_ctx->ratio;
+
+  size_t pix_count = g_cairo_fb.var.xres * g_cairo_fb.var.yres;
+  switch (g_cairo_fb.var.bits_per_pixel)
+  {
+  case 8:
+    p_ctx->fbbuff.c = (uint8_t*)malloc(pix_count * sizeof(uint8_t));
+    break;
+  case 15:
```
Comment (crertel) [inline c_src/device/cairo.c:120]: Which format is this one, OOC?

**Review by crertel — APPROVED** [review]

File: c_src/device/cairo.c (hunk 1 above)
Comment (ringlej) [reply c_src/device/cairo.c:120]: The 15 bit formats are the one that I haven't actually been able to test. I was able to test all the other color depths using `fbset`. The 15 bit formats will be RGB555 or BGR555.

Comment (crertel) [reply c_src/device/cairo.c:120]: Ah cool, thank you.

**Pull request merged**